tests/test_config_reader.py

```diff
@@ -314,7 +314,31 @@ def test_construct_stacks_constructs_stack(
                 "project_path": self.context.project_path,
                 "custom_key": "custom_value",
             },
-            config=ANY,
+            config={
+                "project_path": self.context.project_path,
+                "stack_group_path": "account/stack-group/region",
+                "profile": "account_profile",
+                "project_code": "account_project_code",
+                "region": "region_region",
+                "required_version": ">1.0",
+                "template_bucket_name": "stack_group_template_bucket_name",
+                "custom_key": "custom_value",
+                "dependencies": ["child/level", "top/level"],
+                "template": {"path": "path/to/template"},
+                "parameters": {"param1": "val1"},
+            },
+            vars={
+                "var": {},
+                "project_path": self.context.project_path,
+                "stack_group_path": "account/stack-group/region",
+                "profile": "account_profile",
+                "project_code": "account_project_code",
+                "region": "region_region",
+                "required_version": ">1.0",
+                "template_bucket_name": "stack_group_template_bucket_name",
+                "custom_key": "custom_value",
+                "dependencies": ["top/level"],
+            },
```

Comment:
I guess that a bit of the ugliness of the internal implementation is made more visible here to the user, in the duplication that exists between the self.template_vars structure used during config rendering and then the config itself. 

In a real world example (the values used in these mock setups seem to be a bit contrived), there is less overlap than suggested in the above, but there will still be some overlap (e.g. both will contain keys like 

- j2_environment
- profile
- project_code
- project_path
- region
- stack_group_path
- template_bucket_name
- template_key_prefix